tests/contract/test_resource_client.py

```diff
@@ -670,3 +670,62 @@ def test_is_primary_identifier_equal_fail(resource_client):
         {"a": 1, "b": 2, "c": 3},
         {"a": 1, "b": 2, "c": 4},
     )
+
+
+def test_assert_write_only_property_does_not_exist(resource_client):
+    schema = {
+        "a": {"type": "number", "const": 1},
+        "b": {"type": "number", "const": 2},
+        "c": {"type": "number", "const": 3},
+    }
+    resource_client._update_schema(schema)
+    resource_client.assert_write_only_property_does_not_exist(
+        schema, resource_client.write_only_paths
+    )
+
+
+def test_assert_write_only_property_does_not_exist_success(resource_client):
+    with pytest.raises(KeyError):
+        schema = {
+            "properties": {
+                "a": {"type": "number", "const": 1},
+                "b": {"type": "number", "const": 2},
+                "c": {"type": "number", "const": 3},
+                "d": {"type": "number", "const": 4},
+            },
+            "readOnlyProperties": ["/properties/b"],
+            "createOnlyProperties": ["/properties/c"],
+            "primaryIdentifier": ["/properties/c"],
+            "writeOnlyProperties": ["/properties/d"],
```

Comment:
add a test with multiple writeOnlyProperties

Reply:
do we need a separate test for this? Or just adding it this test will work?

Reply:
i would just parameterize this test with two schemas